### PR TITLE
Update homepage charts and improve performance 

### PR DESCRIPTION
**Pull request by deastman**
# Description

This PR aims to improve page load performance by swapping out large vendor libraries for smaller alternative vendor libraries. 

The specific vendor library swaps are:
- chartjs in lieu of plotly for the charts on the homepage
- fontello in lieu of fontawesome for icons

Since chartjs is a JavaScript library, we now fetch most of the data for the homepage after initial page load via Ajax. 

## (Feature) Demo/Screenshots
Screen Shot 2022-05-19 at 12 49 47 PM

## Important Changes
`va_explorer/home/urls.py`
- Adds an endpoint `/trends`

`va_explorer/home/views.py`
- Adds a class-based view, Trends, which returns a json object with relevant data for populating the statistics and charts on the homepage

`va_explorer/home/va_trends.py`
- Extracts the logic for generating the trends data into a separate module. (Note: I have not changed the core logic used to generate these data since the goal of the PR isn't a refactor.)

`va_explorer/static/js/home.js`
- Makes Ajax call to `/trends` and populate the statistics and charts on the homepage 

`va_explorer/templates/base.html`
- Removes fontawesome and adds fontello

`va_explorer/templates/home/index.html`
- Updates markup to align with new approach for loading chart and statistics data

`va_explorer/templates/partials/_va_table.html`
`va_explorer/templates/va_data_cleanup/index.html`
`va_explorer/templates/va_data_management/edit.html`
`va_explorer/templates/va_data_management/index.html`
`va_explorer/templates/va_data_management/show.html`
`va_explorer/va_analytics/dash_apps/va_dashboard.py`
`va_explorer/va_analytics/utils/rendering.py`
- Update markup to use fontello icons instead of fontawesome

`va_explorer/home/tests/test_views.py`
- Updates tests given new Json endpoint

`requirements/base.txt`
- Adds time-machine to remove date-based variation from getting today's date with respect to testing 

## Testing Recommendations
- Test that the functionality matches previous functionality
- Test that all icons appear as they should (location, appearance, etc.)

### CODE REVIEW

**Review by ajbarnes — CHANGES_REQUESTED** [review]

File: va_explorer/templates/base.html [hunk 1]
```diff
@@ -15,12 +15,17 @@
...
-  <link href="{% static 'vendor/bootstrap-4.3.1/bootstrap.min.css' %}" rel="stylesheet">
-  <link href="{% static 'vendor/select2-4.1.0-beta.1/select2.css' %}" rel="stylesheet">
-  <link href="{% static 'css/form.css' %}" rel="stylesheet">
-  <link href="{% static 'css/project.css' %}" rel="stylesheet">
-  <link href="{% static 'css/fab.css' %}" rel="stylesheet"> {% endblock %} {% load plotly_dash%} {% plotly_header %}
+  <link rel="icon" href="{% static 'images/favicons/favicon.ico' %}">
+  {% block css %}
+    <link href="{% static 'vendor/bootstrap-4.3.1/bootstrap.min.css' %}" rel="stylesheet">
+    <link href="{% static 'vendor/select2-4.1.0-beta.1/select2.css' %}" rel="stylesheet">
+    <link href="{% static 'vendor/fontello-a86a8ad9/css/fontello.css' %}" rel="stylesheet">
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/templates/base.html]: No minified css or version for fontello. If we wanted to use a minified css file, we'd have to minify it ourselves.

Fontello is a website where a developer can handpick icons from different libraries (fontawesome is one such library) and download the relevant font files and css for showing those icons (see: https://fontello.com/) The letters after fontello in the directory name are a hash that just represent the download instance.

File: va_explorer/static/vendor/fontello-a86a8ad9/demo.html [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    }
+    .codesOn .i-code {
+      display: inline;
+    }
+    .i-code {
+      display: none;
+    }
+    @font-face {
+      font-family: 'fontello';
+      src: url('./font/fontello.eot?20711964');
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/static/vendor/fontello-a86a8ad9/demo.html]: Removed demo and css files we don't need: 
[a6be1ff](https://github.com/VA-Explorer/va_explorer/pull/237/commits/a6be1ff4ba5cdb2307b146dac453335d61414e80)

File: va_explorer/static/vendor/fontello-a86a8ad9/demo.html [hunk 3]
```diff
@@ -0,0 +1,241 @@
+<!DOCTYPE html>
+<html>
+  <head>
+  <!--[if lt IE 9]><script language="javascript" type="text/javascript" src="//html5shim.googlecode.com/svn/trunk/html5.js"></script><![endif]-->
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/static/vendor/fontello-a86a8ad9/demo.html]: Removed! [a6be1ff](https://github.com/VA-Explorer/va_explorer/pull/237/commits/a6be1ff4ba5cdb2307b146dac453335d61414e80)

File: va_explorer/static/vendor/fontello-a86a8ad9/css/fontello-ie7.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
+[class^="icon-"], [class*=" icon-"] {
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/static/vendor/fontello-a86a8ad9/css/fontello-ie7.css]: Removed! 
[a6be1ff](https://github.com/VA-Explorer/va_explorer/pull/237/commits/a6be1ff4ba5cdb2307b146dac453335d61414e80)

File: va_explorer/static/vendor/fontello-a86a8ad9/LICENSE.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+Font license info
+
+
+## Font Awesome
+
+   Copyright (C) 2016 by Dave Gandy
+
+   Author:    Dave Gandy
+   License:   SIL ()
+   Homepage:  http://fortawesome.github.com/Font-Awesome/
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/static/vendor/fontello-a86a8ad9/LICENSE.txt]: Yes, license is correct as these are Font Awesome icons that are provided individually through fontello. See earlier comment.

File: va_explorer/static/vendor/fontello-a86a8ad9/css/fontello.css [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+  /* Font smoothing. That was taken from TWBS */
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+
+  /* Uncomment for 3D effect */
+  /* text-shadow: 1px 1px 1px rgba(127, 127, 127, 0.3); */
+}
+
+.icon-up-dir:before { content: '\e800'; } /* '' */
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/static/vendor/fontello-a86a8ad9/css/fontello.css]: Not sure of the intention of those comments. They appear automatically in the downloaded css.

File: va_explorer/static/js/logging_event_handlers.js [hunk 7]
```diff
@@ -47,29 +47,29 @@ $(document).ready(function(){
...
+    $('#trend_tab').click(function() {
         $.ajax({
             url: log_url,
             method: 'POST',
             data: {name: "dashboard", message: "Switched to VA Trends Tab"}
         });
     });
-    $('#demographic_tab').click(function() { 
+    $('#demographic_tab').click(function() {
         console.log('clicked demo tab')
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/static/js/logging_event_handlers.js]: [8b5675c](https://github.com/VA-Explorer/va_explorer/pull/237/commits/8b5675cd3d70b3bd48efeae11a2c79e4162d62df)

File: va_explorer/va_analytics/dash_apps/va_dashboard.py [hunk 8]
```diff
@@ -1402,9 +1401,15 @@ def make_card(
                 [
                     header,
                     html.Span(
-                        html.Span(className="fas fa-info-circle"),
-                        style={"margin-left": "5px", "color": "rgba(75,75,75,0.5)"},
-                        id=f"{card_id}-tooltip-target",
+                        html.I(
+                            className="icon-info-circled",
```
[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/va_analytics/dash_apps/va_dashboard.py:1405]: Sorry for missing this one and thank you for the catch!
[8024f0c](https://github.com/VA-Explorer/va_explorer/pull/237/commits/8024f0c6110620f3f64735c8a1404bd8dbac89b4)

File: va_explorer/templates/base.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajbarnes) [reply va_explorer/templates/base.html]: Should we add documentation for devs on how to update fontello if we ever want to add/remove icons in the future?

[the remark this answers is not in the document]
Comment (deastman) [reply va_explorer/templates/base.html]: Added at long last! [57d6db9](https://github.com/VA-Explorer/va_explorer/pull/237/commits/57d6db93808204f02eef26159336bb8402e75062)

**Pull request merged**